sdk/python/kfp/_client.py

```diff
@@ -559,7 +559,16 @@ def list_pipeline_versions(self, pipeline_id: str, page_token='', page_size=10,
     )
 
   # TODO: provide default namespace, similar to kubectl default namespaces.
-  def run_pipeline(self, experiment_id, job_name, pipeline_package_path=None, params={}, pipeline_id=None, version_id=None):
+  def run_pipeline(
+      self,
+      experiment_id: str,
+      job_name: str,
+      pipeline_package_path: Optional[str] = None,
+      params: Optional[dict] = None,
+      pipeline_id: Optional[str] = None,
+      version_id: Optional[str] = None,
+      pipeline_root: Optional[str] = None,
```

Comment:
nit: may I confirm what's our official name for this parameter? in IR it's called output_directory, see https://github.com/kubeflow/pipelines/blob/d9c019641ef9ebd78db60cdb78ea29b0d9933008/api/v2alpha1/pipeline_spec.proto#L38.

Reply:
I think `pipeline_root` is fine. We have been calling it this way since the beginning, and TFX also uses the same name.
/cc @neuromage

Reply:
SGTM, we might want to change the IR field name to align in the next IR version